src/nanoemoji/color_glyph.py

```diff
@@ -36,11 +36,16 @@
 from typing import Generator, NamedTuple, Optional, Sequence, Tuple
 import ufoLib2
 
-
 def _scale_viewbox_to_emsquare(view_box: Rect, upem: int) -> Tuple[float, float]:
     # scale to font upem
-    return (upem / view_box.w, upem / view_box.h)
-
+    view_box_w = upem / view_box.w
+    view_box_h = upem / view_box.h
+
+    # upem = 1024
+    # Don't zoom when the viewbox width is greater than upem.
+    if view_box.w > upem:
+        view_box_w = 1.0
```

Comment:
but here you are not maintaining the aspect ratio. You're only handling the width independently of the height, and only when viewbox.w exceeds the upem (which is not always 1024 btw).
If you scale the width and height differently you won't preserve the aspect ratio.

Reply:
Yes, I haven't figured out how to keep the aspect ratio while making it overflow. How do we scale the graphics while maintaining the aspect ratio? Do you have any advice?